src/Server/Workspace.php

```diff
@@ -200,4 +214,44 @@ public function xdependencies(): array
         }
         return $dependencyReferences;
     }
+
+    /**
+     * @param Options|null $settings
+     */
+    public function didChangeConfiguration(Options $settings = null)
+    {
+        if ($settings === null) {
+            return;
+        }
+
+        $changedOptions = $this->getChangedOptions($settings);
+        $this->options = $settings;
```

Comment:
Mostly because of side-effects since the server is running until closed.

### Proposals:

1. `Options` trait with method `setOptions` for dependencies
2. `Options` will be handled by `didChangeConfiguration` and they (the values) will be passed as parameter to methods 
> e.g. `$indexer->index($fileTypes, $maxFileSize);`

I like option 1 more then 2 because the `Workspace` does not need to know how the dependencies use them. He only needs to inform them about the new settings.

Either way the `Workspace` needs all classes as dependencies that require `Options`.